### PR TITLE
Do not overwrite report content URI when setting Report to COMPLETED

### PR DESCRIPTION
**Pull request by wtritch**

### CODE REVIEW

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/GeneratePdf.java [hunk 1]
```diff
@@ -34,19 +34,16 @@
     private final ReportGenerationTemporaryStorage reportGenerationTemporaryStorage;
     private final PdfMergeService pdfMerger;
     private final ReportService reportService;
-    private final CollateStudentReportsSink sink;
```
Comment (wtritch) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/GeneratePdf.java]: Unrelated, this no longer needs to be injected.

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/GeneratePdf.java [hunk 2]
```diff
@@ -63,22 +60,21 @@ public void collateReportPdf(final ReportRequestMessage message) {
...
+                    logger.warn("Unable to remove temporary merged report PDF file: {}", mergedPdf);
+                }
 
-            reportGenerationTemporaryStorage.cleanup(message.getReportId());
-            logger.debug("Report: {} complete", message.getReportId());
+                final Report completed = Report.builder()
+                        .copy(writtenReport)
+                        .status(ReportStatus.COMPLETED)
+                        .build();
+                reportService.update(completed);
```
Comment (wtritch) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/GeneratePdf.java]: This was the problem, the reportContentService.writeContent method sets the report's content URI.  I was then promptly clearing it out when updating the report to COMPLETED.

File: report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/step/GeneratePdfTest.java [hunk 3]
```diff
@@ -109,5 +116,9 @@ public void itShouldMergeAndStorePdfContents() throws Exception {
         assertThat(mergedPdfContents.exists()).isFalse();
 
         verify(reportContentService).writeContent(eq(report), any(FileInputStream.class), eq(mergedPdfLength));
+
+        final ArgumentCaptor<Report> reportCaptor = ArgumentCaptor.forClass(Report.class);
+        verify(reportService).update(reportCaptor.capture());
+        assertThat(reportCaptor.getValue().getReportResourceUri()).isEqualTo(URI.create("my_uri"));
```
Comment (wtritch) [inline report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/step/GeneratePdfTest.java]: This test was failing before my changes.

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/GeneratePdf.java (hunk 2 above)
Comment (esotrope) [reply report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/GeneratePdf.java]: Not seeing where it was cleared out. was cleanup() removing it?

Comment (wtritch) [reply report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/GeneratePdf.java]: I was updating and saving the originally-fetched report that didn't yet have a content URI assigned to it.

Comment (bgsavage) [reply report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/GeneratePdf.java]: This reminds me of one of Uncle Bob's rules about output arguments: 
 http://www.informit.com/articles/article.aspx?p=1382188

**Pull request merged**